spec/advance_enumerable_spec.rb

```diff
@@ -0,0 +1,163 @@
+require '../advace_enumerable.rb'
+RSpec.describe Enumerable do
+  include Enumerable
+  let(:arr) { [2, 1, 7] }
+  let(:arr_nil) { [false, nil, false] }
+  let(:multiply) do
+    proc do |x|
+      x * 3
+    end
+  end
+  describe '#my_each' do
+    it 'performs operation on each element of array if block is given' do
+      expect(arr.my_each { |x| puts x }).to eql(arr.each { |x| puts x })
```

Comment:
You should have tests to check if your method works on other types of enumerables besides arrays.